### ISSUE TITLE
Listen / Register to multiple actions

### ISSUE THREAD

**Issue by rafaelchiti** [acdlite/flummox#69]
Would be nice to have the ability to register a few actions at a time and more importantly I think to have a generic 'matcher' mechanism. ( @acdlite you mentioned that there is already work in progress in the area, just wanted to share ideas and also keep updated on how the impl is going to be)

Lets say I want to create a matcher that basically matches any action that has a status code 500.

I would like to have a store that does that in a simple way:

```js

var matcher = (payload) => payload.response.status === '500';

class NotificationsStore extends Store {

  constructor(flux) {
    this.registerMatcher(matcher, handler);
  }

}

```

Another cool thing would be to define metadata in the action, therefore we could do a matcher that matches against that. Lets say we want to 'tag' all our navigation actions with a tag 'navigation'
then I could build a matcher in the way `(payload) => payload.action.tags.contains('navigation')`
and listen to every navigation action.

Thanks! really liking the library and concepts, awesome work!

**Comment by acdlite** [comment]
Thanks opening this issue! Plan is to add `Store#registerAll()` for registering a handler for all actions.

Re: metadata, that seems outside the scope of Flummox, especially since it would be fairly easy for the user to implement. What do you find lacking in the standard approach of dispatching an object with certain fields?

```js
// in actions class
someAction() {
  ...
  return {
    actionType: 'navigation',
    foo: 'bar';
  };
}

// in store class
handleAll(payload) {
  if (payload.actionType === 'navigation') {...}
}
```

**Comment by acdlite** [comment]
In other words I'm hesitant to add this to the core library, and instead offer the tools for other libraries to implement on top of Flummox by extending the base Store class.

**Comment by acdlite** [comment]
e.g.

```js
const matchingMixin =  {
  registerMatcher(matcher, handler) {
    this.registerAll(payload => {
      if (matcher(payload)) {
        handler();
      }
    });
  }
}

class MyStore extends Store {...}
Object.assign(MyStore.prototype, matchingMixin);
```

Although now that I've typed that out... that does seem like a very natural API. Hmm, maybe this does belong in the core :D

**Comment by acdlite** [comment]
Added `Store#registerMatch(matcher, handler)` in 4.0 branch. Released as part of 4.0.0-alpha.
